### ISSUE TITLE
[Bug]: [Nightly]Kafka runs much slowly and will be killed for 6h timeout with error `panic: close of closed channel`

### ISSUE THREAD

**Issue by NicoYuan1986** [milvus-io/milvus#25101]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 72c5e2a
- Deployment mode(standalone or cluster):  cluster
- MQ type(rocksmq, pulsar or kafka):     kafka
- SDK version(e.g. pymilvus v2.0.0rc2):2.4.0.dev75
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Kafka runs much slowly and will be killed for 6h timeout with error `panic: close of closed channel`.

pymilvus error:
```
[2023-06-24 22:47:40,454 - ERROR - ci_test]: Traceback (most recent call last):
  File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 26, in inner_wrapper
    res = func(*args, **_kwargs)
  File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 57, in api_request
    return func(*arg, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 443, in insert
    res = conn.batch_insert(self._name, entities, partition_name,
  File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 109, in handler
    raise e
  File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 105, in handler
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 136, in handler
    ret = func(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 85, in handler
    raise e
  File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 50, in handler
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 481, in batch_insert
    raise err
  File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 477, in batch_insert
    raise MilvusException(response.status.error_code, response.status.reason)
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=Local: Unknown partition)>
 (api_request.py:39)
[2023-06-24 22:47:40,454 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=Local: Unknown partition)> (api_request.py:40)
```

milvus error:
```
2023-06-25T06:15:33.251827708+08:00 stderr F panic: close of closed channel
2023-06-25T06:15:33.25184743+08:00 stderr F
2023-06-25T06:15:33.251852992+08:00 stderr F goroutine 1221 [running]:
2023-06-25T06:15:33.251856696+08:00 stderr F panic({0x3ac09e0, 0x4403a40})
2023-06-25T06:15:33.251861179+08:00 stderr F    /usr/local/go/src/runtime/panic.go:941 +0x397 fp=0xc007043678 sp=0xc0070435b8 pc=0x14d03f7
2023-06-25T06:15:33.25186542+08:00 stderr F runtime.closechan(0xc002528ba0)
2023-06-25T06:15:33.251876323+08:00 stderr F    /usr/local/go/src/runtime/chan.go:364 +0x3dc fp=0xc0070436b8 sp=0xc007043678 pc=0x149c2bc
2023-06-25T06:15:33.25188013+08:00 stderr F github.com/confluentinc/confluent-kafka-go/kafka.(*Consumer).Close(0xc00d271200)
2023-06-25T06:15:33.251890141+08:00 stderr F    /go/pkg/mod/github.com/confluentinc/confluent-kafka-go@v1.9.1/kafka/consumer.go:421 +0x2c fp=0xc0070436f0 sp=0xc0070436b8 pc=0x27cd7ec
2023-06-25T06:15:33.251896766+08:00 stderr F github.com/milvus-io/milvus/pkg/mq/msgstream/mqwrapper/kafka.(*Consumer).Close.func1()
2023-06-25T06:15:33.25190057+08:00 stderr F     /go/src/github.com/milvus-io/milvus/pkg/mq/msgstream/mqwrapper/kafka/kafka_consumer.go:255 +0x45 fp=0xc007043710 sp=0xc0070436f0 pc=0x27e5a05
2023-06-25T06:15:33.251905656+08:00 stderr F sync.(*Once).doSlow(0xc007043790?, 0x14a6228?)
2023-06-25T06:15:33.251909286+08:00 stderr F    /usr/local/go/src/sync/once.go:68 +0xc2 fp=0xc007043770 sp=0xc007043710 pc=0x1512f02
2023-06-25T06:15:33.251913037+08:00 stderr F sync.(*Once).Do(...)
2023-06-25T06:15:33.251916494+08:00 stderr F    /usr/local/go/src/sync/once.go:59
2023-06-25T06:15:33.251923577+08:00 stderr F github.com/milvus-io/milvus/pkg/mq/msgstream/mqwrapper/kafka.(*Consumer).Close(0x3f61250?)
2023-06-25T06:15:33.251927578+08:00 stderr F    /go/src/github.com/milvus-io/milvus/pkg/mq/msgstream/mqwrapper/kafka/kafka_consumer.go:252 +0x45 fp=0xc0070437a0 sp=0xc007043770 pc=0x27e5985
2023-06-25T06:15:33.251934048+08:00 stderr F github.com/milvus-io/milvus/pkg/mq/msgstream.(*mqMsgStream).Close(0xc00d271140)
2023-06-25T06:15:33.251938634+08:00 stderr F    /go/src/github.com/milvus-io/milvus/pkg/mq/msgstream/mq_msgstream.go:202 +0x33c fp=0xc007043940 sp=0xc0070437a0 pc=0x2b773fc
2023-06-25T06:15:33.251942621+08:00 stderr F github.com/milvus-io/milvus/pkg/mq/msgstream.(*MqTtMsgStream).Close(0xc00d77de00)
2023-06-25T06:15:33.251950422+08:00 stderr F    /go/src/github.com/milvus-io/milvus/pkg/mq/msgstream/mq_msgstream.go:542 +0x32 fp=0xc007043958 sp=0xc007043940 pc=0x2b7a872
2023-06-25T06:15:33.251963327+08:00 stderr F github.com/milvus-io/milvus/pkg/mq/msgdispatcher.(*Dispatcher).Handle.func1()
2023-06-25T06:15:33.251973954+08:00 stderr F    /go/src/github.com/milvus-io/milvus/pkg/mq/msgdispatcher/dispatcher.go:186 +0x11b fp=0xc0070439c8 sp=0xc007043958 pc=0x2d8a6fb
2023-06-25T06:15:33.251977773+08:00 stderr F sync.(*Once).doSlow(0xc0043e1b00?, 0x3f25275?)
2023-06-25T06:15:33.251987443+08:00 stderr F    /usr/local/go/src/sync/once.go:68 +0xc2 fp=0xc007043a28 sp=0xc0070439c8 pc=0x1512f02
2023-06-25T06:15:33.251999099+08:00 stderr F sync.(*Once).Do(...)
2023-06-25T06:15:33.252004339+08:00 stderr F    /usr/local/go/src/sync/once.go:59
2023-06-25T06:15:33.252016042+08:00 stderr F github.com/milvus-io/milvus/pkg/mq/msgdispatcher.(*Dispatcher).Handle(0xc00095c680?, 0x3?)
2023-06-25T06:15:33.2520453+08:00 stderr F      /go/src/github.com/milvus-io/milvus/pkg/mq/msgdispatcher/dispatcher.go:184 +0x41d fp=0xc007043b50 sp=0xc007043a28 pc=0x2d8a4dd
2023-06-25T06:15:33.25205358+08:00 stderr F github.com/milvus-io/milvus/pkg/mq/msgdispatcher.(*dispatcherManager).tryMerge(0xc0027c7950)
2023-06-25T06:15:33.252072642+08:00 stderr F    /go/src/github.com/milvus-io/milvus/pkg/mq/msgdispatcher/manager.go:216 +0x625 fp=0xc007043e08 sp=0xc007043b50 pc=0x2d8d6c5
2023-06-25T06:15:33.25207827+08:00 stderr F github.com/milvus-io/milvus/pkg/mq/msgdispatcher.(*dispatcherManager).Run(0xc0027c7950)
2023-06-25T06:15:33.252081952+08:00 stderr F    /go/src/github.com/milvus-io/milvus/pkg/mq/msgdispatcher/manager.go:170 +0x3f2 fp=0xc007043fc8 sp=0xc007043e08 pc=0x2d8ce12
2023-06-25T06:15:33.252091401+08:00 stderr F github.com/milvus-io/milvus/pkg/mq/msgdispatcher.(*client).Register.func2()
2023-06-25T06:15:33.252096646+08:00 stderr F    /go/src/github.com/milvus-io/milvus/pkg/mq/msgdispatcher/client.go:72 +0x2b fp=0xc007043fe0 sp=0xc007043fc8 pc=0x2d8846b
2023-06-25T06:15:33.252104398+08:00 stderr F runtime.goexit()
2023-06-25T06:15:33.252118802+08:00 stderr F    /usr/local/go/src/runtime/asm_amd64.s:1571 +0x1 fp=0xc007043fe8 sp=0xc007043fe0 pc=0x1505a61
2023-06-25T06:15:33.252124264+08:00 stderr F created by github.com/milvus-io/milvus/pkg/mq/msgdispatcher.(*client).Register
2023-06-25T06:15:33.252134797+08:00 stderr F    /go/src/github.com/milvus-io/milvus/pkg/mq/msgdispatcher/client.go:72 +0x492
2023-06-25T06:15:33.252150179+08:00 stderr F
2023-06-25T06:15:33.252163714+08:00 stderr F goroutine 1 [chan receive]:
2023-06-25T06:15:33.252176194+08:00 stderr F runtime.gopark(0xc0002e9760?, 0xc0002e97b8?, 0xf3?, 0x31?, 0xc0002e97b8?)
2023-06-25T06:15:33.252190385+08:00 stderr F    /usr/local/go/src/runtime/proc.go:361 +0xd6 fp=0xc001cb3788 sp=0xc001cb3768 pc=0x14d3656
2023-06-25T06:15:33.252199468+08:00 stderr F runtime.chanrecv(0xc0003e7980, 0xc0002e9c58, 0x1)
2023-06-25T06:15:33.252209124+08:00 stderr F    /usr/local/go/src/runtime/chan.go:577 +0x56c fp=0xc001cb3818 sp=0xc001cb3788 pc=0x149c8ec
2023-06-25T06:15:33.252222155+08:00 stderr F runtime.chanrecv1(0xc0003e7980?, 0xc0002e9da8?)
2023-06-25T06:15:33.252234641+08:00 stderr F    /usr/local/go/src/runtime/chan.go:440 +0x18 fp=0xc001cb3840 sp=0xc001cb3818 pc=0x149c318
2023-06-25T06:15:33.252263423+08:00 stderr F github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run(0xc0002e9e48, 0x0, {0xc0009580b0?, 0xc?})
2023-06-25T06:15:33.252274734+08:00 stderr F    /go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:351 +0xa07 fp=0xc001cb3df8 sp=0xc001cb3840 pc=0x3574e27
2023-06-25T06:15:33.25229898+08:00 stderr F github.com/milvus-io/milvus/cmd/milvus.(*run).execute(0xc000956270, {0xc00013c060?, 0x3, 0x3}, 0xc000952480)
2023-06-25T06:15:33.252312362+08:00 stderr F    /go/src/github.com/milvus-io/milvus/cmd/milvus/run.go:117 +0x6ae fp=0xc001cb3ee0 sp=0xc001cb3df8 pc=0x358154e
2023-06-25T06:15:33.252332837+08:00 stderr F github.com/milvus-io/milvus/cmd/milvus.RunMilvus({0xc00013c060?, 0x3, 0x3})
2023-06-25T06:15:33.252358125+08:00 stderr F    /go/src/github.com/milvus-io/milvus/cmd/milvus/milvus.go:60 +0x21e fp=0xc001cb3f58 sp=0xc001cb3ee0 pc=0x3580d9e
2023-06-25T06:15:33.252370673+08:00 stderr F main.main()
2023-06-25T06:15:33.252406126+08:00 stderr F    /go/src/github.com/milvus-io/milvus/cmd/main.go:26 +0x2e fp=0xc001cb3f80 sp=0xc001cb3f58 pc=0x3583c2e
2023-06-25T06:15:33.252424782+08:00 stderr F runtime.main()
2023-06-25T06:15:33.252432421+08:00 stderr F    /usr/local/go/src/runtime/proc.go:250 +0x212 fp=0xc001cb3fe0 sp=0xc001cb3f80 pc=0x14d3292
2023-06-25T06:15:33.252436221+08:00 stderr F runtime.goexit()
2023-06-25T06:15:33.252444065+08:00 stderr F    /usr/local/go/src/runtime/asm_amd64.s:1571 +0x1 fp=0xc001cb3fe8 sp=0xc001cb3fe0 pc=0x1505a61
2023-06-25T06:15:33.252448991+08:00 stderr F
2023-06-25T06:15:33.252452819+08:00 stderr F goroutine 2 [force gc (idle)]:

...
```

### Expected Behavior

pass

### Milvus Log

1. link: https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI/detail/master/413/pipeline/205/
2. log: [artifacts-milvus-distributed-kafka-nightly-413-pymilvus-e2e-logs.tar.gz](https://jenkins.milvus.io:18080/job/Milvus%20Nightly%20CI/job/master/413/artifact/artifacts-milvus-distributed-kafka-nightly-413-pymilvus-e2e-logs.tar.gz)

**Comment by NicoYuan1986** [comment]
The issue has occurred for four consecutive days. So add the urgent label.

**Comment by jaime0815** [comment]
related to https://github.com/milvus-io/milvus/issues/25065

**Comment by jaime0815** [comment]
it had been fixed on 2.2.0 branch

**Comment by NicoYuan1986** [comment]
Fixed. master-20230625-7b999b42
